cassiopeia/tools/small_parsimony.py

```diff
@@ -0,0 +1,230 @@
+"""
+This file contains several tools useful for using small-parsimony to analyze
+phylogenies.
+
+Amongst these tools are basic Fitch-Hartigan reconstruction, parsimony scoring,
+and the FitchCount algorithm described in Quinn, Jones et al, Science (2021).
+"""
+from cassiopeia.mixins.errors import CassiopeiaTreeError
+from typing import Optional
+
+import numpy as np
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.mixins import CassiopeiaError
```

Comment:
Could we combine the `CassiopeiaError` and `CassiopeiaTreeError` imports?

Reply:
Done.